pcsx2/CDVD/CDVDcommon.cpp

```diff
@@ -303,6 +303,22 @@ static CDVD_SourceType m_CurrentSourceType = CDVD_SourceType::NoDisc;
 
 void CDVDsys_SetFile(CDVD_SourceType srctype, std::string newfile)
 {
+#ifdef WIN32
+	if (Path::IsAbsolute(newfile))
+	{
+		const auto& splitPath = Path::SplitNativePath(newfile);
+		// GetDriveType() Requires trailing backslashes
+		const auto root = fmt::format("{}\\", splitPath.at(0));
+
+		const auto driveType = GetDriveType(StringUtil::UTF8StringToWideString(root).c_str());
+		if (driveType == DRIVE_REMOVABLE)
+		{
+			Host::AddOSDMessage("Warning! Game disc location is on a removable drive, performance issues such as jittering and freezing may occur.", 5.0f);
```

Comment:
Probably best to use `Host::AddIconOSDMessage()` here, that way it doesn't duplicate if you start two games quickly after another (you'll need to include `IconsFontAwesome.h`). Also don't need the double console log, all OSD messages also get written to the log now.

For example:
```
Host::AddIconOSDMessage("RemovableDriveWarning", ICON_FA_EXCLAMATION_TRIANGLE, "Game disc location is on a removable drive, performance issues such as jittering and freezing may occur.", Host::OSD_WARNING_DURATION);
```

Otherwise LGTM.

Reply:
Sweet, I've done that instead.